core/src/main/scala/org/apache/spark/sql/store/CodeGeneration.scala

```diff
@@ -260,6 +283,32 @@ object CodeGeneration extends Logging {
         "dialect")).asInstanceOf[GeneratedStatement]
   }
 
+
+  private[this] def compileGeneratedIndexUpdate(table: String,
```

Comment:
code generation is required to set each parameter of the table. like in citi, if table has 250 columns, setting each column of an index otherwise will incur lots of switch case runtime evaluation. this have direct bearing on performance of CachedBatch creation and IMO is required.